WordPress/src/main/java/org/wordpress/android/models/Role.java

```diff
@@ -0,0 +1,95 @@
+package org.wordpress.android.models;
+
+import android.support.annotation.StringRes;
+
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+
+public enum Role {
+    ADMIN(R.string.role_admin),
+    EDITOR(R.string.role_editor),
+    AUTHOR(R.string.role_author),
+    CONTRIBUTOR(R.string.role_contributor),
+    FOLLOWER(R.string.role_follower),
+    VIEWER(R.string.role_viewer);
+
+    private final int mLabelResId;
+
+    Role(@StringRes int labelResId) {
+        mLabelResId = labelResId;
+    }
+
+    public String toDisplayString() {
+        return WordPress.getContext().getString(mLabelResId);
+    }
+
+    public static Role fromString(String role) {
+        switch (role) {
+            case "administrator":
+                return ADMIN;
+            case "editor":
+                return EDITOR;
+            case "author":
+                return AUTHOR;
+            case "contributor":
+                return CONTRIBUTOR;
+            case "follower":
+                return FOLLOWER;
+            case "viewer":
+                return VIEWER;
+        }
+        throw new IllegalArgumentException("All roles must be handled");
```

Comment:
The only case this will be thrown is if there is a developer error and it'll help catch it early. That's why I added it, as it should never reach there in a user's device. Otherwise we can return `null` but will need to do a null-check everywhere. What do you think?